Adding more icons for extensions and filenames - and some other changes

Babkock opened the pull request:
Hello, I hope this pull request finds you all well. First of all, I want to say I am excited about this project. I had opened a pull request on the now-abandoned [exa](https://github.com/ogham/exa) that was never really responded to. In the past I have contributed to the file browser **[nnn](https://github.com/jarun/nnn)**, and the other active `ls` clone that is an alternative to eza and exa, **[lsd](https://github.com/lsd-rs/lsd)**. In my first commit here, I added some icons for the extensions `zig`, `xbps`, and `scpt`. I also added some other common filenames, that you seemed to be missing, to the icons list for exact filename matches. You were missing "`.doom.d`", "`.emacs.d`", "`meson.build`", "`config.el`", and others. My second commit is where I make subjective icon changes: I changed the PHP icon to the elephant (`\u{e608}`), and I changed the ASM/assembly icon to the pound sign (`\u{f0423}`).

I also noticed that you were using a generic lightning bolt symbol for the Zig programming language, when there is an official Zig Unicode icon in the Nerd Fonts (`\u{e6a9}`). I have fixed that for you.

It is okay if you want to keep the PHP and ASM icons how they are, that's why I put them in the second and third commits, so I can roll back if necessary. But I strongly prefer the look of the ones I added. I also want to ask how you all feel about maybe switching to the Go gopher icon, instead of the word "GO"? Just feeling things out. It is totally fine, and I would not be offended if you do not wish to go in that direction.

Thank you all very much for your time. I am open to making more commits, to change anything you would like me to change. I also have ideas for other subjective icon changes, if you are interested.

cafkafk commented on src/output/icons.rs:
```diff
@@ -699,7 +717,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "yaml"           => Icons::YAML,             // 
     "yml"            => Icons::YAML,             // 
     "z"              => Icons::COMPRESSED,       // 
-    "zig"            => '\u{21af}',              // ↯
+    "zig"            => Icons::LANG_ZIG,         // 
```
We should obviously go with the official zig logo

cafkafk commented on src/output/icons.rs:
```diff
@@ -687,6 +704,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "wmv"            => Icons::VIDEO,            // 
     "woff"           => Icons::FONT,             // 
     "woff2"          => Icons::FONT,             // 
+    "xbps"           => '\u{f32e}',              // 
```
This seems to be the void linux logo for a file for void linux, makes sense

cafkafk commented on src/output/icons.rs:
```diff
@@ -604,9 +620,10 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
+    "s"              => Icons::LANG_ASSEMBLY,    // 
     "sass"           => Icons::LANG_SASS,        // 
     "scala"          => '\u{e737}',              // 
+    "scpt"           => Icons::OS_APPLE,         // 
```
this is the apple logofor an applescript file, makes sense

cafkafk commented on src/output/icons.rs:
```diff
@@ -604,9 +620,10 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "rtf"            => Icons::TEXT,             // 
     "ru"             => Icons::LANG_RUBY,        // 
     "rubydoc"        => Icons::LANG_RUBYRAILS,   // 
-    "s"              => Icons::LANG_ASSEMBLY,    // 
+    "s"              => Icons::LANG_ASSEMBLY,    // 
```
I'm not sure about this one tbh, \ue614 is for css

Babkock replied:
Hi, is "s" an actual CSS file extension? Because I associate it with assembly language files. To be more specific I think they appear more often with a single capital "S" (like "`bootloader.S`"), so I could fix that. Like a C compiler will give you a ".S" assembly file if you do this: `$ cc -S program.c` -> `program.S`

cafkafk commented on src/output/icons.rs:
```diff
@@ -544,15 +560,15 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "out"            => '\u{eb2c}',              // 
     "p12"            => Icons::KEY,              // 
     "par"            => Icons::COMPRESSED,       // 
-    "part"           => '\u{f43a}',              // 
+    "part"           => Icons::DOWNLOAD,         // 󰇚
     "patch"          => Icons::DIFF,             // 
     "pbm"            => Icons::IMAGE,            // 
     "pdf"            => '\u{f1c1}',              // 
     "pem"            => Icons::KEY,              // 
     "pfx"            => Icons::KEY,              // 
     "pgm"            => Icons::IMAGE,            // 
-    "phar"           => Icons::LANG_PHP,         // 
-    "php"            => Icons::LANG_PHP,         // 
+    "phar"           => Icons::LANG_PHP,         // 
+    "php"            => Icons::LANG_PHP,         // 
```
I'm also not sure about this one

PThorpe92 replied:
as someone who regularly must look at these file extensions... I approve of the elephant

cafkafk commented on src/output/icons.rs:
```diff
@@ -544,15 +560,15 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "out"            => '\u{eb2c}',              // 
     "p12"            => Icons::KEY,              // 
     "par"            => Icons::COMPRESSED,       // 
-    "part"           => '\u{f43a}',              // 
+    "part"           => Icons::DOWNLOAD,         // 󰇚
```
Honestly, I'm not sure I like either of these, but I'm leaning towards the latter

cfxegbert replied:
This is consistent with our other partiality downloaded files

cafkafk commented on src/output/icons.rs:
```diff
@@ -394,6 +409,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "ejs"            => '\u{e618}',              // 
     "el"             => Icons::EMACS,            // 
     "elc"            => Icons::EMACS,            // 
+    "elf"            => Icons::OS_LINUX,         // 
```
This makes perfect sense

cafkafk commented on src/output/icons.rs:
```diff
@@ -316,7 +331,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "ar"             => Icons::COMPRESSED,       // 
     "arw"            => Icons::IMAGE,            // 
     "asc"            => Icons::SHIELD_LOCK,      // 󰦝
-    "asm"            => Icons::LANG_ASSEMBLY,    // 
+    "asm"            => Icons::LANG_ASSEMBLY,    // 󰐣
```
I'm not sure about this one tbh, \ue614 is for css

cafkafk commented on src/output/icons.rs:
```diff
@@ -283,6 +297,7 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "requirements.txt"    => Icons::LANG_PYTHON,    // 
     "robots.txt"          => '\u{f06a9}',           // 󰚩
     "rubydoc"             => Icons::LANG_RUBYRAILS, // 
+    "rustfmt.toml"        => Icons::LANG_RUST,      // 
```
Yeop, fine with me

cafkafk commented on src/output/icons.rs:
```diff
@@ -266,13 +279,14 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "Makefile.in"         => Icons::MAKE,           // 
     "MANIFEST"            => Icons::LANG_PYTHON,    // 
     "MANIFEST.in"         => Icons::LANG_PYTHON,    // 
+    "meson.build"         => Icons::WRENCH,         // 
     "npm-shrinkwrap.json" => Icons::NPM,            // 
     "npmrc"               => Icons::NPM,            // 
     "package-lock.json"   => Icons::NPM,            // 
     "package.json"        => Icons::NPM,            // 
     "passwd"              => Icons::LOCK,           // 
     "PKGBUILD"            => '\u{f303}',            // 
-    "php.ini"             => Icons::LANG_PHP,       // 
+    "php.ini"             => Icons::LANG_PHP,       // 
```
again not sure about these

cafkafk commented on src/output/icons.rs:
```diff
@@ -266,13 +279,14 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "Makefile.in"         => Icons::MAKE,           // 
     "MANIFEST"            => Icons::LANG_PYTHON,    // 
     "MANIFEST.in"         => Icons::LANG_PYTHON,    // 
+    "meson.build"         => Icons::WRENCH,         // 
```
Hmm, sure

cafkafk commented on src/output/icons.rs:
```diff
@@ -246,10 +256,13 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "Jenkinsfile"         => '\u{e66e}',            // 
     "jsconfig.json"       => Icons::LANG_JAVASCRIPT,// 
     "Justfile"            => Icons::WRENCH,         // 
+    "Kbuild"              => Icons::OS_LINUX,       // 
+    "Kconfig"             => Icons::OS_LINUX,       // 
```
yeop, makes sense

cafkafk commented on src/output/icons.rs:
```diff
@@ -246,10 +256,13 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "id_ed25519"          => Icons::PRIVATE_KEY,    // 󰌆
     "id_ed25519_sk"       => Icons::PRIVATE_KEY,    // 󰌆
     "id_rsa"              => Icons::PRIVATE_KEY,    // 󰌆
+    "init.el"             => Icons::EMACS,          // 
```
makes sense

cafkafk commented on src/output/icons.rs:
```diff
@@ -222,6 +231,7 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "Dockerfile"          => Icons::DOCKER,         // 
     "Earthfile"           => '\u{f0ac}',            // 
     "environment"         => Icons::CONFIG,         // 
+    "fstab"               => Icons::DATABASE,       // 
```
I don't like this one tbh

cafkafk commented on src/output/icons.rs:
```diff
@@ -205,9 +211,12 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "config"              => Icons::CONFIG,         // 
+    "config.el"           => Icons::EMACS,          // 
+    "config.in"           => Icons::CONFIG,         // 
+    "config.mk"           => Icons::MAKE,           // 
```
this makes sense

cafkafk commented on src/output/icons.rs:
```diff
@@ -205,9 +211,12 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
+    "composer.lock"       => Icons::LANG_PHP,       // 
     "config"              => Icons::CONFIG,         // 
+    "config.el"           => Icons::EMACS,          // 
+    "config.in"           => Icons::CONFIG,         // 
```
sure

cafkafk commented on src/output/icons.rs:
```diff
@@ -205,9 +211,12 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
+    "composer.json"       => Icons::LANG_PHP,       // 
+    "composer.lock"       => Icons::LANG_PHP,       // 
     "config"              => Icons::CONFIG,         // 
+    "config.el"           => Icons::EMACS,          // 
```
okay

cafkafk commented on src/output/icons.rs:
```diff
@@ -205,9 +211,12 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "Cargo.lock"          => Icons::LANG_RUST,      // 
     "Cargo.toml"          => Icons::LANG_RUST,      // 
     "CMakeLists.txt"      => Icons::MAKE,           // 
-    "composer.json"       => Icons::LANG_PHP,       // 
-    "composer.lock"       => Icons::LANG_PHP,       // 
+    "composer.json"       => Icons::LANG_PHP,       // 
+    "composer.lock"       => Icons::LANG_PHP,       // 
```
again, not sure about these

MartinFillon replied:
I agree with @cafkafk even tho those are useful files in php if not in a php context those icons don't make any sense.

PThorpe92 replied:
Seeing as there are two diff options for php icons.. perhaps use one for composer and another for .php files?

cafkafk commented on src/output/icons.rs:
```diff
@@ -174,6 +179,7 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     ".kshrc"              => Icons::SHELL,          // 󱆃
     ".login"              => Icons::SHELL,          // 󱆃
     ".logout"             => Icons::SHELL,          // 󱆃
+    ".nix-channels"       => '\u{f313}',            // 
```
yes

cafkafk commented on src/output/icons.rs:
```diff
@@ -140,6 +144,7 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
     "npm_cache"           => Icons::FOLDER_NPM,     // 
     "pam.d"               => Icons::FOLDER_KEY,     // 󰢬
     "Pictures"            => '\u{f024f}',           // 󰉏
+    "src"                 => Icons::XML,            // 󰗀
```
sure ish, might be to generic thou?

cafkafk replied:
is this too generic

cafkafk commented on src/output/icons.rs:
```diff
@@ -114,8 +115,11 @@ impl Icons {
+    ".emacs.d"            => Icons::EMACS,          // 
     ".git"                => Icons::FOLDER_GIT,     // 
     ".github"             => Icons::FOLDER_GITHUB,  // 
+    ".gnupg"              => Icons::FOLDER_KEY,     // 󰢬
```
hmm fine

cafkafk commented on src/output/icons.rs:
```diff
@@ -114,8 +115,11 @@ impl Icons {
 /// all the directories that have a custom icon.
 const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
     ".config"             => Icons::FOLDER_CONFIG,  // 
+    ".doom.d"             => Icons::EMACS,          // 
+    ".emacs.d"            => Icons::EMACS,          // 
```
also okay

cafkafk commented on src/output/icons.rs:
```diff
@@ -68,6 +68,7 @@ impl Icons {
     const LANG_STYLUS: char     = '\u{e600}';  // 
     const LANG_TEX: char        = '\u{e69b}';  // 
     const LANG_TYPESCRIPT: char = '\u{e628}';  // 
+    const LANG_ZIG: char        = '\u{e6a9}';  // 
```
yes

cafkafk commented on src/output/icons.rs:
```diff
@@ -58,7 +58,7 @@ impl Icons {
     const LANG_OCAML: char      = '\u{e67a}';  // 
     const LANG_PERL: char       = '\u{e67e}';  // 
-    const LANG_PHP: char        = '\u{e73d}';  // 
+    const LANG_PHP: char        = '\u{e608}';  // 
```
not sure

cafkafk commented on src/output/icons.rs:
```diff
@@ -45,7 +45,7 @@ impl Icons {
     const KEY: char             = '\u{eb11}';  // 
     const KEYPASS: char         = '\u{f23e}';  // 
-    const LANG_ASSEMBLY: char   = '\u{e637}';  // 
+    const LANG_ASSEMBLY: char   = '\u{f0423}'; // 󰐣
```
not sure

Babkock replied:
Changed assembly icon

cafkafk reviewed (commented):
Don't change anything yet I'm just writing down how I feel about each of them

cafkafk commented on src/output/icons.rs:
```diff
@@ -45,7 +45,7 @@ impl Icons {
     const KEY: char             = '\u{eb11}';  // 
     const KEYPASS: char         = '\u{f23e}';  // 
-    const LANG_ASSEMBLY: char   = '\u{e637}';  // 
+    const LANG_ASSEMBLY: char   = '\u{f0423}'; // 󰐣
```
This one needs to be discussed

cafkafk commented on src/output/icons.rs:
```diff
@@ -58,7 +58,7 @@ impl Icons {
     const LANG_OCAML: char      = '\u{e67a}';  // 
     const LANG_PERL: char       = '\u{e67e}';  // 
-    const LANG_PHP: char        = '\u{e73d}';  // 
+    const LANG_PHP: char        = '\u{e608}';  // 
```
This one needs to be discussed

Babkock replied:
I think the elephant is more visually appealing than a logo with tiny letters that just says "PHP". Same thing with the assembly language icon - it doesn't have to be the pound sign, but it should be better than a glyph that just says "ASM".

cafkafk commented on src/output/icons.rs:
```diff
@@ -222,6 +231,7 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "Dockerfile"          => Icons::DOCKER,         // 
     "Earthfile"           => '\u{f0ac}',            // 
     "environment"         => Icons::CONFIG,         // 
+    "fstab"               => Icons::DATABASE,       // 
```
I am not sure I like this one, the icon makes me think of a binary format, not the actual plain text fstab

```
# <device>                                <dir> <type> <options> <dump> <fsck>
UUID=0a3407de-014b-458b-b5c1-848e92a327a3 /     ext4   defaults  0      1
UUID=f9fe0b69-a280-415d-a03a-a32752370dee none  swap   defaults  0      0
UUID=b411dc99-f0a0-4c87-9e05-184977be8539 /home ext4   defaults  0      2
```

Babkock replied:
It's all good, maybe we could go with generic config gear `\u{e615}`, or the hard disk icon `\u{f0a0}`? I was just thinking `fstab` -> filesystem -> disk

cafkafk reviewed (changes requested):
These are the ones I think should be discussed

MartinFillon commented on src/output/icons.rs:
```diff
@@ -283,11 +306,13 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "shadow"              => Icons::LOCK,           // 
     "shells"              => Icons::CONFIG,         // 
     "sudoers"             => Icons::LOCK,           // 
+    "template"            => '\u{f32e}',            // 
```
this one could be matched on the `.template` files too

Babkock replied:
Composer is the unofficial package manager for PHP. Maybe `composer.json` appears in non-PHP projects sometimes, but not often.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -58,7 +59,7 @@ impl Icons {
     const LANG_KOTLIN: char     = '\u{e634}';  // 
     const LANG_OCAML: char      = '\u{e67a}';  // 
     const LANG_PERL: char       = '\u{e67e}';  // 
-    const LANG_PHP: char        = '\u{e73d}';  // 
```
PHP does not use the elephant as the current logo.

Babkock replied:
It might not be official, but the elephant mascot is well-known and appears when you Google image search "php". I personally prefer the PHP elephant, and I know others do, too. Maybe if someone else says they don't want this, I can change it. Other projects such as `lsd`, [ranger_devicons](https://github.com/alexanderjeurissen/ranger_devicons) (I've contributed to both), [vim-devicons](https://github.com/ryanoasis/vim-devicons/tree/master) and [all-the-icons.el for Emacs](https://github.com/domtronn/all-the-icons.el/blob/master/logo.png) use the elephant for PHP, so it is not unheard of, and it would make `eza` more consistent with these other projects.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -113,12 +115,19 @@ impl Icons {
     ".npm"                => Icons::FOLDER_NPM,     // 
     ".ssh"                => Icons::FOLDER_KEY,     // 󰢬
     ".Trash"              => '\u{f1f8}',            // 
+    "bin"                 => Icons::FOLDER_CONFIG,  // 
```
bin is not a configuration folder

Babkock replied:
Sorry it seems like a common folder that people would have

cfxegbert commented on src/output/icons.rs:
```diff
@@ -139,11 +148,15 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
     "pam.d"               => Icons::FOLDER_KEY,     // 󰢬
+    "portage"             => '\u{f30d}',            // 
     "Pictures"            => '\u{f024f}',           // 󰉏
+    "root"                => '\u{f0250}',           // 󰉐
```
Why would root be a lock icon?

Babkock replied:
Why not? It is a folder at the root of the filesystem that normal users can not access.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -139,11 +148,15 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
+    "portage"             => '\u{f30d}',            // 
     "Pictures"            => '\u{f024f}',           // 󰉏
+    "root"                => '\u{f0250}',           // 󰉐
+    "src"                 => '\u{f19fc}',           // 󱧼
```
Not sure why source code would be a wrench folder.

Babkock replied:
I guess I was thinking wrench -> build -> source code. If you don't like it I can change it, or we can go with a different icon.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -139,11 +148,15 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
+    "src"                 => '\u{f19fc}',           // 󱧼
     "ssh"                 => Icons::FOLDER_KEY,     // 󰢬
     "sudoers.d"           => Icons::FOLDER_KEY,     // 󰢬
+    "vendor"              => '\u{f0ae6}',           // 󰫦
```
How is vendor specific to Symfony?

Babkock replied:
Symfony is not as associated with Composer as I once thought.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -139,11 +148,15 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
+    "vendor"              => '\u{f0ae6}',           // 󰫦
     "Videos"              => '\u{f03d}',            // 
-    "xbps.d"              => Icons::FOLDER_CONFIG,  // 
+    "xbps.d"              => '\u{f32e}',            // 
```
That is not a directory type icon.  We have been trying to use directory icons.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -198,16 +212,20 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     ".zsh_sessions"       => Icons::SHELL,          // 󱆃
     "._DS_Store"          => Icons::OS_APPLE,       // 
     "a.out"               => Icons::SHELL_CMD,      // 
-    "authorized_keys"     => '\u{f08c0}',           // 󰣀
```
This is a SSH specific and should not be changed to a license icon.

Babkock replied:
I just thought keys -> keys icon

cfxegbert commented on src/output/icons.rs:
```diff
@@ -246,11 +265,14 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
-    "known_hosts"         => '\u{f08c0}',           // 󰣀
+    "Kbuild"              => Icons::OS_LINUX,       // 
+    "Kconfig"             => Icons::OS_LINUX,       // 
+    "known_hosts"         => '\u{e60a}',            // 
```
This is a SSH specific and should not be changed to a license icon.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -283,11 +306,13 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "shadow"              => Icons::LOCK,           // 
     "shells"              => Icons::CONFIG,         // 
     "sudoers"             => Icons::LOCK,           // 
+    "template"            => '\u{f32e}',            // 
```
Template does not seem void linux specific.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -429,16 +456,16 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "gv"             => '\u{f1049}',             // 󱁉
     "gz"             => Icons::COMPRESSED,       // 
-    "h"              => Icons::LANG_C,           // 
+    "h"              => Icons::HEADER,           // 
```
We just changed these to language specific files.  We would not want to revert these back to header.

Babkock replied:
Oh sorry. It does not have to have its own constant, but I like the "H" icon. I will change if others don't want.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -642,6 +672,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "svelte"         => '\u{e697}',              // 
     "svg"            => Icons::VECTOR,           // 󰕙
     "swift"          => '\u{e755}',              // 
+    "sym"            => '\u{eae8}',              // 
```
Not sure about this.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -670,6 +701,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "tzo"            => Icons::COMPRESSED,       // 
     "unity"          => Icons::UNITY,            // 
     "unity3d"        => Icons::UNITY,            // 
+    "url"            => '\u{f0ac}',              // 
```
Seems like \u{f1fa} could be a better choice.

Babkock replied:
I'm not sure why, the "@" icon seems specific to email for some reason. Maybe someone else can weigh in on this one.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -685,8 +717,10 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "windows"        => Icons::OS_WINDOWS,       // 
     "wma"            => Icons::AUDIO,            // 
     "wmv"            => Icons::VIDEO,            // 
+    "webloc"         => '\u{f0ac}',              // 
```
Seems like \u{f1fa} could be a better choice.  Should make a constant if using multiple times.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -198,16 +212,20 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "config"              => Icons::CONFIG,         // 
+    "config.el"           => Icons::EMACS,          // 
+    "config.in"           => Icons::CONFIG,         // 
+    "config.mk"           => Icons::MAKE,           // 
```
Should this be added to FileType::Build?

cfxegbert commented on src/output/icons.rs:
```diff
@@ -266,13 +288,14 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "Makefile.in"         => Icons::MAKE,           // 
     "MANIFEST"            => Icons::LANG_PYTHON,    // 
     "MANIFEST.in"         => Icons::LANG_PYTHON,    // 
+    "meson.build"         => Icons::WRENCH,         // 
```
Should be added to FileType::Build

cfxegbert commented on src/output/icons.rs:
```diff
@@ -390,10 +415,12 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "dylib"          => Icons::OS_APPLE,         // 
     "ebook"          => Icons::BOOK,             // 
     "ebuild"         => '\u{f30d}',              // 
+    "eclass"         => '\u{f30d}',              // 
```
Make a constant

cfxegbert commented on src/output/icons.rs:
```diff
@@ -537,22 +565,23 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "p12"            => Icons::KEY,              // 
     "par"            => Icons::COMPRESSED,       // 
-    "part"           => '\u{f43a}',              // 
+    "part"           => Icons::DOWNLOAD,         // 󰇚
```
Needs to be added to FileType::Temp

Babkock commented:
Hello, where are we on this? Sorry, I did not see the `FileType` enums or implementation. I have made the necessary changes in that file (`src/info/filetype.rs`). I just want to weigh in on the elephant? It seems like some of us (myself included) like the look of the PHP elephant association, and others don't. I can change it if it seems that more people don't like it. How should we proceed with this pull request? I will make any necessary changes that the community wants.

cfxegbert commented on src/output/icons.rs:
```diff
@@ -113,12 +122,19 @@ impl Icons {
 /// Mapping from full filenames to directory icon. This mapping should contain
 /// all the directories that have a custom icon.
 const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
+    "__pycache__"         => Icons::LANG_PYTHON,    // 󰌠
```
We have to decide if we want non-folder icons for __pycache__, .cargo, .doom.d, and .emacs.d

Babkock replied:
I understand why others might not want this, because they are out of place in the directory icons. I can change this.

MartinFillon replied:
I think icons for those folder is good

cfxegbert replied:
We have decided we do not want icons for these unless they are folder icons.

Babkock replied:
Sounds good

MartinFillon commented:
Could you handle #560 @Babkock please ?

zzhaolei commented:
Would it be better to change the symlink icon to "\u{f481}"?

f481:
<img width="23" alt="image" src="https://github.com/eza-community/eza/assets/17561793/f1ba8c1b-a661-40c5-8bd6-4433b68e995e">


before:
<img width="498" alt="image" src="https://github.com/eza-community/eza/assets/17561793/69d1359a-b942-4699-828d-e016125ef61d">

after:
<img width="484" alt="image" src="https://github.com/eza-community/eza/assets/17561793/5cba96af-b16c-4df3-b289-9f554fc4b729">

cfxegbert reviewed (changes requested):
Please address my comments

Babkock commented:
Hello, here is an update. I have added some code to `pub fn icon_for_file()`. I have addressed @zzhaolei 's request, and implemented code to show different icons for symbolic links. I have added `Icons::SYMLINK` and `Icons::SYMLINK_DIR` constants. I have also added some code (at least on Unix operating systems) that will show the `Icons::SHELL_CMD` icon if a file has no extension, and is marked executable. I have achieved this with the functions `.is_link()` and `.is_executable_file()`. I also used the exact icons used by zzhaolei in his screenshot.

How does this look? If there is a better way to do this, please let me know. My knowledge of Rust is very rudimentary and basic, I don't know all the tricks. I figured this would be better than having the generic file icon for these types of files.

MartinFillon reviewed (changes requested):
Code is fine with me but the commit history is not
